smtp/smtpc.h

```diff
@@ -116,6 +116,7 @@ struct smtpc_request {
 	int status;
 	/** Error message. */
 	const char *reason;
+	char *error;
```

Comment:
We have `reason` that is commented as 'error message' and `error` without any comment (is not it an 'error message'?). It does not look descriptive enough. Let's think about proper comments together.

First, what reader's questions the comments should answer?

1. How the field is intended to use, what to expect here in the general case, what are corner cases (say, a value at request failure).
2. How to manage memory pointed by pointers.

So we should highlight that the `reason` field is always filled in `smtpc_execute()` if it returns zero and is `NULL` otherwise. We should clarify that it is a string to report details of an error to a user. We should mention that it does not require freeing (points to a static buffer or to `error`).

The `error` is actually a temporary buffer that should exist during a request lifetime. It is not expected to be read directly, because `reason` points to it, when appropriate. It is always non-NULL and may contain zeros or a zero terminated error message depending on how the request is failed. It must be freed at freeing the request structure (but not before that to don't corrupt `reason`).

I would rename the `error` field: since it is just buffer, maybe `error_buf` will reflect its usage better?

Reply:
Thank you for explanation! I've renamed the field and added some comments there that explain it.

Reply:
Those comments resolve the main problem: not it is clear what are `reason` and `error_buf`.

Several nits:

> It takes the value of -1 if SMTP code doesn't match to error status.

I don't understand the 'code doesn't match to error status' phrase, to be honest.

There is no information that `reason` can be `NULL` (and when).

Reply:
Fixed.

Reply:
I looked again and, it seems, I was not completely right. I'll cite the new part of your comment:

> * It can be NULL in some exceptional situations,
> * when there is some Curl internal memory issue
> * or returned Curl code is unexpected.

But in those cases the request is deleted and accessing this field is illegal anyway. So we can state that it is defined and never NULL if `smtpc_execute()` retuns zero.

Reply:
Fixed.

Reply:
Thanks!